site_tools/image_deployer.py

```diff
@@ -0,0 +1,315 @@
+            use_object_store_creds=False,
+            image_prefix="testing_",
+            image_type="qcow2",
+            dry_run=False):
+    logging.info(f"Syncing images (dry_run={dry_run})...")
+    for available_image in available_images:
+        if should_sync_image(available_image.disk_name, site_images, current):
+            sync_image(
+                object_connection,
+                image_connection,
+                available_image,
+                current=current,
+                use_object_store_creds=use_object_store_creds,
+                image_prefix=image_prefix,
+                image_type=image_type,
+                dry_run=dry_run
+            )
+    logging.info("Sync completed.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Deploy images to the site.")
+
+    parser.add_argument("--site-yaml", type=str, required=True,
+                        help="A yaml file with site information for syncing.")
+    parser.add_argument("--supports-yaml", type=str,
+                        default="/etc/chameleon_image_tools/supports.yaml",
+                        help="A yaml file with supported images.")
+    parser.add_argument("--dry-run",
+                        action="store_true",
+                        help="Perform a dry run without making any changes.")
+    # TODO(pdmars): add a force sync flag that overrides the current check
+
+    # TODO(pdmars): probably just have these for testing
+    # final version will be public only containers, but for now I'm using my account
+    parser.add_argument("--use-object-store-creds", action="store_true",
+                        help="Use creds with the object store for the image store.")
+    parser.add_argument("--filter-public-visibility", action="store_true",
+                        help="The visibility to filter when listing images.")
+
+    args = parser.parse_args()
+
+    if args.dry_run:
+        logging.info("Dry run mode enabled. No changes will be made.")
+
+    # TODO: add this stuff in?
+    # loop over supported_distros to figure out release and variant
+    #with open(args.supports_yaml, 'r') as f:
+    #    supports = yaml.safe_load(f)
+
+    with open(args.site_yaml, "r") as f:
+        site = yaml.safe_load(f)
+
+    base_container = site.get("image_container", "chameleon-images")
+    scope = site.get("scope", "prod")
+    image_type = site.get("image_type", "qcow2")
+    image_prefix = site.get("image_prefix", "testing_")
+    object_store_cloud = site.get("object_store_cloud", "chi_uc") # TODO: change default
+    image_store_cloud = site.get("image_store_cloud", "uc_dev") # TODO: change default
+
+    logging.debug(f"Using base image container/scope: {base_container}/{scope}")
+    object_connection = get_openstack_connection(object_store_cloud)
+    image_connection = get_openstack_connection(image_store_cloud)
+
+    current = get_current_value(object_connection, base_container, scope)
```

Comment:
object_connection and get_current_value are still requiring a valid clouds.yaml entry

Reply:
Yeah, I guess I should move from testing this behind credentials on my account to public stuff so we can just remove all that